StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift

```diff
@@ -72,14 +74,30 @@ final class PaymentSheetLoader {
                             intent: intent
                         )
                     }
+
+                // Determine if Link and Apple Pay are enabled
                 let isLinkEnabled = isLinkEnabled(intent: intent, configuration: configuration)
-                analyticsClient.logPaymentSheetEvent(event: .paymentSheetLoadSucceeded,
-                                                                     duration: Date().timeIntervalSince(loadingStartDate))
+                let isApplePayEnabled = StripeAPI.deviceSupportsApplePay()
+                    && configuration.applePay != nil
+                    && intent.isApplePayEnabled
+
+                // Send load finished analytic
+                // This is hacky; the logic to determine the default selected payment method belongs to the SavedPaymentOptionsViewController. We invoke it here just to report it to analytics before that VC loads.
+                let (defaultSelectedIndex, paymentOptionsViewModels) = SavedPaymentOptionsViewController.makeViewModels(
```

Comment:
Does the default pm need to be sent in the loaded event? Seems like we have access to an initialized `SavedPaymentOptionsViewController` in `PaymentSheetViewController` and `PaymentSheetFlowControllerViewController`, which is where we also log the show event. Can we attach the default to the show event? Would that be less hacky?

Reply:
I think so. For FlowController it's possible (likely, even, if you have a default PM) to never show the sheet before completing payment.